x-pack/legacy/plugins/infra/public/pages/metrics/index.tsx

```diff
@@ -114,6 +114,10 @@ export const MetricDetail = withMetricPageProviders(
           []
         );
 
+        if (metadataLoading && !filteredLayouts.length) {
+          return null;
```

Comment:
I looked at this using the testing (slow) cluster and I think instead of returning `null` we should probably return a loading screen (similar to the one deeper in the tree). The reason being is that while it's waiting on the `metadata` request, you're staring at a blank screen for a few seconds.